python/lsst/ctrl/mpexec/preExecInit.py

```diff
@@ -172,3 +182,42 @@ def saveInitOutputs(self, graph):
                     # butler will raise exception if dataset is already there
                     _LOG.debug("Saving InitOutputs for task=%s key=%s", task, name)
                     self.butler.put(initOutputVar, attribute.name, {})
+
+    def saveConfigs(self, graph):
+        """Write configurations for pipeline tasks to butler or check that
+        existing configurations are equal to the new ones.
+
+        Parameters
+        ----------
+        graph : `~lsst.pipe.base.QuantumGraph`
+            Execution graph.
+
+        Raises
+        ------
+        Exception
+            Raised if ``skipExisting`` is `False` and datasets already
+            exists. Content of a butler collection may be changed if
+            exception is raised.
```

Comment:
We could make this exception safe by wrapping it in `with self.butler.transaction()`, I think.

Reply:
Indeed, let me try that.

Reply:
That works fine with ci_hsc_gen3, will add that.